docs/proposals/resource-quota.md

```diff
@@ -0,0 +1,404 @@
+  // operator is "In", and the values array contains only "value". The requirements are ANDed.
+  MatchFields map[string]string `json:"matchFields,omitempty"`
+  // matchExpressions is a list of field selector requirements. The requirements are ANDed.
+  MatchExpressions []FieldSelectorRequirement `json:"matchExpressions,omitempty"`
+}
+
+// A field selector requirement is a selector that contains values, a fieldPath, and an operator that
+// relates the fieldPath and values.
+type FieldSelectorRequirement struct {
+  // fieldPath is the path that the selector applies to.
+  FieldPath string `json:"fieldPath" patchStrategy:"merge" patchMergeKey:"fieldPath"`
+  // operator represents a fieldPath's relationship to a set of values.
+  // Valid operators ard In, NotIn, Exists and DoesNotExist.
+  Operator FieldSelectorOperator `json:"operator"`
+  // values is an array of string values. If the operator is In or NotIn,
+  // the values array must be non-empty. If the operator is Exists or DoesNotExist,
+  // the values array must be empty. This array is replaced during a strategic
+  // merge patch.
+  Values []string `json:"values,omitempty"`
+}
+
+// A field selector operator is the set of operators that can be used in a selector requirement.
+type FieldSelectorOperator string
+
+const (
+  FieldSelectorOpIn           FieldSelectorOperator = "In"
+  FieldSelectorOpNotIn        FieldSelectorOperator = "NotIn"
+  FieldSelectorOpExists       FieldSelectorOperator = "Exists"
+  FieldSelectorOpDoesNotExist FieldSelectorOperator = "DoesNotExist"
+)
+```
+
+### ResourceQuota
+
+The `ResourceQuotaSpec` will be modified to support the optional
+selectors to filter the `ResourceQuota` document.
+
+```
+// ResourceQuotaSpec defines the desired hard limits to enforce for Quota
+type ResourceQuotaSpec struct {
+  // Hard is the set of desired hard limits for each named resource
+  Hard ResourceList `json:"hard,omitempty"`
+  // LabelSelector is a label query that must match over each resource that is tracked by quota.
+  // If empty, the quota matches every resource.
+  // More info: http://releases.k8s.io/HEAD/docs/user-guide/labels.md#label-selectors
+  LabelSelector *LabelSelector `json:"labelSelector,omitempty"`
+  // FieldSelector is a field query that must match over the resource tracked by quota.
+  // If empty, the quota matches every resource.
+  FieldSelector *FieldSelector `json:"fieldSelector,omitempty"`
+}
+```
+
+The selectors are immutable post-creation.
+
+API definition of `ReplenishmentPolicy` to support replenishment
+via a rate are deferred pending further prototyping to understand
+what additional fields may or may not be needed to track when
+quota was last replenished.
+
+### Pod
+
+The administrator is able to filter terminating vs non-terminating workloads
+by defining a field selector of `spec.restartPolicy`.
+
+The administrator is not able to filter pods by quality of service.  To
```

Comment:
QoS is derived from resource request/requirement, so I think it would only make sense to make it immutable if resource request/requirement are immutable. If you don't like using labels for QoS (which is unfortunate from the standpoint of MetadataPolicy... I need to think about what we're going to do about that) then it seems like it would make more sense to put it in PodStatus rather than PodSpec. Brian had suggested this possibility once; you could still have it set by the server at creation time (I remember that was what Brian had in mind).